internal/backend/runtime/omni/infraprovider/state.go

```diff
@@ -389,3 +426,54 @@ func (st *State) checkNamespaceAndType(ns resource.Namespace, infraProviderID st
 	return false, status.Errorf(codes.PermissionDenied, "namespace not allowed, must be one of %s or %s",
 		resources.InfraProviderNamespace, infraProviderSpecificNamespace)
 }
+
+// resourceConfig defines the resource-specific configuration to be validated by the state.
+type resourceConfig struct {
+	readOnlyForProviders bool
+	ephemeral            bool
+	checkID              bool
+}
+
+// IsInfraProviderResource returns true if the given resource type is an infra provider specific resource.
+func IsInfraProviderResource(ns resource.Namespace, resType resource.Type) bool {
+	_, isInfraProviderResource := getResourceConfig(ns, resType)
+
+	return isInfraProviderResource
+}
+
+// getResourceConfig returns the configuration for the given resource type.
+func getResourceConfig(ns resource.Namespace, resType resource.Type) (config resourceConfig, isInfraProviderResource bool) {
+	if strings.HasPrefix(ns, resources.InfraProviderSpecificNamespacePrefix) {
+		return config, true
+	}
+
+	switch resType {
+	case infra.MachineRequestType:
+		return resourceConfig{
+			readOnlyForProviders: true,
+		}, true
+	case infra.MachineRequestStatusType:
+		return resourceConfig{}, true
+	case infra.InfraMachineType:
+		return resourceConfig{
+			readOnlyForProviders: true,
+		}, true
+	case infra.InfraMachineStateType:
+		return resourceConfig{}, true
+	case infra.InfraMachineStatusType:
+		return resourceConfig{}, true
+	case infra.InfraProviderStatusType:
+		return resourceConfig{
+			checkID: true,
+		}, true
+	case infra.InfraProviderHealthStatusType:
+		return resourceConfig{
+			ephemeral: true,
+			checkID:   true,
+		}, true
+	case infra.ConfigPatchRequestType:
+		return resourceConfig{}, true
+	default:
+		return resourceConfig{}, false
+	}
```

Comment:
This is now the central place where we define all configurations for the infra provider resources.

Reply:
Overall it looks good to me, but I don't like that for the normal resources it's easy to guess: ephemeral namespace -- goes to the ephemeral state, default -- goes to the etcd.
Here we use the same namespace.

I would prefer to use the same schema everywhere.